### PR TITLE
Add dev container

### PR DESCRIPTION
**Pull request by BloodFutur**
This adds a Development container configuration for our project. 

Basically, here are the things done:
- It creates a Docker container with ROS2 installed
- It loads a configuration files when running the container
- When it is loaded for the first time (or modified), it tries to build with `colcon` our ROS2 nodes
- Documentation so that you know how to use these files

Feel free to ask questions about it if you do not understand why or how it works.

### CODE REVIEW

**Review by kekoleko123 — COMMENTED** [review]
FOR WINDOWS USER : 
You have to comment the line 53 of the .devcontainer/devcontainer.json file (source=/tmp/.X11-unix,....)

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      "-p9090:9090",
+      "-e", "DISPLAY=${env:DISPLAY}",
+      "--network=host",
+      "--name=geicar",
+      "-v=/dev:/dev",
+      // "--runtime=nvidia", // NVIDIA GPU support
+      "--device=/dev/dri:/dev/dri" // Intel GPU support
+    ],
+    "mounts": [
+      "source=/tmp/.X11-unix,target=/tmp/.X11-unix,type=bind,consistency=cached",
```
Comment (kekoleko123) [inline .devcontainer/devcontainer.json:53]: You have to comment this line if you are a windows user.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      "-p9090:9090",
+      "-e", "DISPLAY=${env:DISPLAY}",
+      "--network=host",
+      "--name=geicar",
+      "-v=/dev:/dev",
+      // "--runtime=nvidia", // NVIDIA GPU support
+      "--device=/dev/dri:/dev/dri" // Intel GPU support
+    ],
+    "mounts": [
+      //"source=/tmp/.X11-unix,target=/tmp/.X11-unix,type=bind,consistency=cached",
```
Comment (kekoleko123) [inline .devcontainer/devcontainer.json:53]: khgmkwfh

File: .devcontainer/postCreateCommand.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+sudo apt update && \
+sudo apt install ros-humble-rviz2 -y && \
```
Comment (Soso0400) [inline .devcontainer/postCreateCommand.sh:4]: super

**Review by emmacstn — APPROVED** [review]
good job :)

**Review by Soso0400 — APPROVED** [review]
Very good Ronan!

**Review by tanghuongcam — COMMENTED** [review]
approved

**Pull request merged**